### PR TITLE
Update release process documentation.

### LINKED ISSUE #2083: Release process documentation needs fixing
```
help2man -n "RPC client for the Zcash daemon" src/zcash-cli > contrib/DEBIAN/manpages/zcash-cli.1
help2man -n "Network daemon for interacting with the Zcash blockchain" src/zcashd > contrib/DEBIAN/manpages/zcashd.1
```

1. Folder name is debian in repo, not DEBIAN
2. version number in man page includes commit id e.g. 1.0.6-xxxxxxx and needs to be manually cleaned up.

Later on:
```
dch -v $DEBVERSION -D jessie -c contrib/DEBIAN/changelog
```
3. Folder name is debian in repo, not DEBIAN

### PR DESCRIPTION
**Pull request by bitcartel**
Closes #2083, #2084, #2088.

### CODE REVIEW

File: doc/release-process.md [hunk 1]
```diff
@@ -37,11 +37,12 @@ previous release:
...
 
-    Build and commit to update versions, and then perform the following commands:
+Build and commit to update versions, and then perform the following commands:
 
-    help2man -n "RPC client for the Zcash daemon" src/zcash-cli > contrib/DEBIAN/manpages/zcash-cli.1
-    help2man -n "Network daemon for interacting with the Zcash blockchain" src/zcashd > contrib/DEBIAN/manpages/zcashd.1
+    help2man -n "RPC client for the Zcash daemon" src/zcash-cli > contrib/debian/manpages/zcash-cli.1
+    help2man -n "Network daemon for interacting with the Zcash blockchain" src/zcashd > contrib/debian/manpages/zcashd.1
 
+Check the version number in the man pages as they use the commit id e.g. 1.0.6-xxxxxxx which may need to be manually cleaned up.
```
Comment (daira) [inline doc/release-process.md]: Also check the titles use "zcashd" and "zcash-cli", not "zcash".

File: doc/release-process.md [hunk 2]
```diff
@@ -56,22 +57,23 @@ In `configure.ac` and `clientversion.h`:
 
 - Change `CLIENT_VERSION_IS_RELEASE` to false while Zcash is in beta-test phase.
 
-If this release changes the behavior of the protocol or fixes a serious bug, we may
-also wish to change the `PROTOCOL_VERSION` in `version.h`.
-
```
Comment (str4d) [inline doc/release-process.md]: Why is this being removed?

Comment (bitcartel) [inline doc/release-process.md]: I didn't change it.  I think this PR just needs to be rebased to include the commit which introduced that text:
https://github.com/zcash/zcash/commit/0d317cd522ff04796b3a9ecb5ef58cca44d3920b

File: doc/release-process.md (hunk 1 above)
Comment (bitcartel) [inline doc/release-process.md]: Done

File: doc/release-process.md (hunk 2 above)
Comment (bitcartel) [inline doc/release-process.md]: Done.

**Comment by bitcartel** [comment]
@daira @str4d Rebased, push force. Requested changes addressed.

**Comment by str4d** [comment]
Both comments addressed.

@zkbot r+

**Pull request merged**